src/routes/RootRoute.tsx

```diff
@@ -14,8 +14,7 @@ const RootRouteInner = () => {
   const { isDarkMode } = usePageThemeContext();
 
   useEffect(() => {
-    // applyTheme(isDarkMode ? Theme.dark : Theme.light);
-    applyTheme(Theme.light);
+    applyTheme(isDarkMode ? Theme.dark : Theme.light);
```

Comment:
Hey @CzarekDryl great start on this! Would it still work if you completely got rid of this `useEffect` and instead call `applyTheme` in the `changeIsDarkMode` callback instead?

```js
const changeIsDarkMode = useCallback(() => {
  setIsDarkMode((prev) => !prev);
  // applyTheme(...)
}, [setIsDarkMode]);
```